go-controller/pkg/util/ovs.go

```diff
@@ -216,22 +250,28 @@ func RunOVNNbctlUnix(args ...string) (string, string, error) {
 func RunOVNNbctlWithTimeout(timeout int, args ...string) (string, string,
 	error) {
 	var cmdArgs []string
-	if config.OvnNorth.Scheme == config.OvnDBSchemeSSL {
-		cmdArgs = []string{
-			fmt.Sprintf("--private-key=%s", config.OvnNorth.PrivKey),
-			fmt.Sprintf("--certificate=%s", config.OvnNorth.Cert),
-			fmt.Sprintf("--bootstrap-ca-cert=%s", config.OvnNorth.CACert),
-			fmt.Sprintf("--db=%s", config.OvnNorth.GetURL()),
-		}
-	} else if config.OvnNorth.Scheme == config.OvnDBSchemeTCP {
-		cmdArgs = []string{
-			fmt.Sprintf("--db=%s", config.OvnNorth.GetURL()),
+
+	envVars := getNbctlSockPathAsEnvVars()
+	// if OVN_NB_DAEMON environment variable is set, then you cannot use
+	// --db CLI option
+	if len(envVars) == 0 {
```

Comment:
@girishmg I actually meant move this block into getNbctlSockPathAsEnvVars() too. So the full function would be:

```
func getNbctlArgsAndEnv(scheme OvnDBScheme, timeout int) ([]string, []string, error) {
	var env, args []string{}

	args = append(args, fmt.Sprintf("--timeout=%d", timeout))

	if config.Default.NbctlDaemonMode {
		pid, err := ioutil.ReadFile("/var/run/openvswitch/ovn-nbctl.pid")
		if err == nil {
			envVars = append(envVars,
				fmt.Sprintf("OVN_NB_DAEMON=/var/run/openvswitch/ovn-nbctl.%s.ctl",
					strings.Trim(string(pid), " \n")))
			logrus.Debugf("using ovn-nbctl daemon mode at %s", envVars)
		} else {
			logrus.Warningf("failed to retrieve ovn-nbctl daemon's control socket " +
				"so resorting to non-daemon mode")
		}
	} else {
		if config.OvnNorth.Scheme == config.OvnDBSchemeSSL {
			args = append(args,
				fmt.Sprintf("--private-key=%s", config.OvnNorth.PrivKey),
				fmt.Sprintf("--certificate=%s", config.OvnNorth.Cert),
				fmt.Sprintf("--bootstrap-ca-cert=%s", config.OvnNorth.CACert),
				fmt.Sprintf("--db=%s", config.OvnNorth.GetURL()))
			}
		} else if config.OvnNorth.Scheme == config.OvnDBSchemeTCP {
			args = append(args, fmt.Sprintf("--db=%s", config.OvnNorth.GetURL()))
			}
		}
	}

	return args, env, nil
}
```

Reply:
Will do. Thanks.